Make JREInstr to work with instrumentation plugins

shurymury opened the pull request:
Additional plugin functionality to allow JREInstr to work.
Tests.

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must not contain extraneous whitespace

### Reviewers
 * [Leonid Kuskov](https://openjdk.java.net/census#lkuskov) (@lkuskov - Committer) ⚠️ Review applies to 22cbd77636175b134c5b6a44d7037d1be25d091c


### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jcov pull/17/head:pull/17` \
`$ git checkout pull/17`

Update a local copy of the PR: \
`$ git checkout pull/17` \
`$ git pull https://git.openjdk.java.net/jcov pull/17/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17`

View PR using the GUI difftool: \
`$ git pr show -t 17`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jcov/pull/17.diff">https://git.openjdk.java.net/jcov/pull/17.diff</a>

</details>

shurymury committed:
An instrumentation plugin should also specify what is to be implanted.

shurymury committed:
Removed unused imports, old version of a test.

lkuskov commented on test/unit/com/sun/tdk/jcov/instrument/plugin/jreinstr/Collect.java:
```diff
@@ -0,0 +1,40 @@
+public class Collect {
+    public static void collect(Object param) {
+
+    }
```
It isn't clear for what this class looking like interface is needed

lkuskov commented:
LGFM

shurymury committed:
Removed empty test method
added time logging for the instrumentation
limited the instrumentation to java.base

shurymury commented:
/integrate
